### ISSUE TITLE
Is WeakValueDictionary serializable?

### ISSUE THREAD

**Issue by djhoese** [cloudpipe/cloudpickle#462]
I have a couple different classes that I'd like to be serialize so I can use them in a dask distributed setting. My current hiccup is with my use of `WeakValueDictionary` and it not be serializable. I tried googling and don't see anyone with this issue specifically which is a little surprising.

Test code:

```python
from cloudpickle import dumps
from weakref import WeakValueDictionary

class MyTest:
    def __init__(self):
        self._cache = WeakValueDictionary()

t = MyTest()
dumps(t)
```

```python-traceback
Traceback (most recent call last):
  File "/home/davidh/repos/git/satpy/check_weakvaluedict_pickle.py", line 9, in <module>
    dumps(t)
  File "/home/davidh/miniconda3/envs/satpy_py39/lib/python3.9/site-packages/cloudpickle/cloudpickle_fast.py", line 73, in dumps
    cp.dump(obj)
  File "/home/davidh/miniconda3/envs/satpy_py39/lib/python3.9/site-packages/cloudpickle/cloudpickle_fast.py", line 602, in dump
    return Pickler.dump(self, obj)
TypeError: cannot pickle 'weakref' object
```

**Comment by djhoese** [comment]
Interestingly, if I move the `_cache` to a class attribute and use `pickle` it works. If I do the same thing (class attribute) but use cloudpickle it fails.

**Comment by djhoese** [comment]
After a lot of debugging I discovered that stdlib pickled handles WeakValueDictionary objects just fine *if* they are defined at the class level and not as an instance property. I also discovered that pickling an instance of a class defined in the `__main__` script/namespace is not a smart idea.

By using `distributed.protocol.serialize` and moving my WeakValueDictionary to a class property I was able to serialize my object.
